src/components/SigninPopup/SigninPopup.js

```diff
@@ -1,72 +1,56 @@
 import React from "react";
 import PopupWithForm from "../PopupWithForm/PopupWithForm";
-import ValidationContext from '../../contexts/ValidationContext';
 
 export default function EditProfilePopup(props) {
-    const errorMessages = React.useContext(ValidationContext);
-    const [email, setEmail] = React.useState("");
-    const [password, setPassword] = React.useState("");
-    const [emailError, setEmailError] = React.useState('');
-    const [passwordError, setPasswordError] = React.useState('');
+    const [values, setValues] = React.useState({
+        email: '',
+        password: '',
+    });
+    const [errors, setErrors] = React.useState({
```

Comment:
**Might be better**

If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you **don’t need** to make functions for handling inputs, storing values, resetting the form. Everything is already done.